packages/react/src/transactions/useCollect.ts

```diff
@@ -1,47 +1,28 @@
-import { createCollectRequest, PublicationFragment } from '@lens-protocol/api-bindings';
+import {
+  createCollectRequest,
+  ProfileOwnedByMeFragment,
+  PublicationFragment,
+} from '@lens-protocol/api-bindings';
 import {
   PendingSigningRequestError,
   UserRejectedError,
   WalletConnectionError,
 } from '@lens-protocol/domain/entities';
-import { invariant } from '@lens-protocol/shared-kernel';
-import { useState } from 'react';
 
-import { useActiveProfile } from '../profile';
+import { Operation, useOperation } from '../helpers';
 import { useCollectController } from './adapters/useCollectController';
 
 export type UseCollectArgs = {
+  collector: ProfileOwnedByMeFragment;
   publication: PublicationFragment;
 };
 
-export function useCollect({ publication }: UseCollectArgs) {
-  const [error, setError] = useState<
-    PendingSigningRequestError | UserRejectedError | WalletConnectionError | null
-  >(null);
-  const [isPending, setIsPending] = useState(false);
-  const collect = useCollectController();
-  const { data: activeProfile } = useActiveProfile();
-
-  return {
-    collect: async () => {
-      try {
-        invariant(activeProfile, 'You must be logged in to collect');
-        setError(null);
-        setIsPending(true);
+export type CollectOperation = Operation<
+  void,
+  PendingSigningRequestError | UserRejectedError | WalletConnectionError
+>;
 
-        const result = await collect(createCollectRequest(publication, activeProfile));
-
-        if (result.isFailure()) {
-          setError(result.error);
-        }
-      } finally {
-        setIsPending(false);
-      }
-    },
-    error,
-    isPending,
-  };
+export function useCollect({ collector, publication }: UseCollectArgs): CollectOperation {
```

Comment:
I think `collector` should be provided as hook arg but publication as `execute` arg, so the same collector can collect different publications, as in `useCreatePost`

Reply:
as a general rule, we can think about it as an `actor` executing an `action` where actor props we define on a hook but action props we define when calling `execute`.

so collector(profile) is collecting(publication)

Reply:
> I think `collector` should be provided as hook arg but publication as `execute` arg, so the same collector can collect different publications, as in `useCreatePost`

The challenge with passing `publication` as callback arg is that the hook has a single `isPending` flag. Without further changes `isPending` will become true on collecting of the first publication and probably disabling the collect button for the entire list.

Sure we can engineer a different strategy to report progress on a case by case (e.g. an "is pending" callback), but why introduce 2 patterns to achieve the same thing if all is need is just nesting `useCollect` closer to the individual list item?